doc/DHCPv6_Relay/DHCPv6_Relay_HLD.md

```diff
@@ -0,0 +1,155 @@
+# DHCP Relay for IPv6 HLD
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+- [DHCP Relay for IPv6 HLD](#dhcp-relay-for-ipv6-hld)
+- [High Level Design Document](#high-level-design-document)
+      - [Rev 0.1](#rev-01)
+- [Table of Contents](#table-of-contents)
+- [List of Tables](#list-of-tables)
+- [List of Figures](#list-of-figures)
+- [Revision](#revision)
+- [About this Manual](#about-this-manual)
+- [Scope](#scope)
+- [Definitions/Abbreviation](#definitionsabbreviation)
+- [1 Requirements Overview](#1-requirements-overview)
+  - [1.1 Functional requirements](#11-functional-requirements)
+  - [1.2 Configuration and Management Requirements](#12-configuration-and-management-requirements)
+- [2 Modules design](#2-modules-design)
+  - [2.1 DHCP Relay for IPv6 build and runtime dependencies](#21-dhcp-relay-for-ipv6-build-and-runtime-dependencies)
+  - [2.2 DHCP Relay for IPv6 process in dhcp-relay container](#22-dhcp-relay-for-ipv6-process-in-dhcp-relay-container)
+  - [2.3 DHCP Monitor](#23-dhcp-monitor)
+- [3 CLI](#3-cli)
+- [4 Init flow](#4-init-flow)
+
+# List of Tables
+* [Table 1: Abbreviations](#definitionsabbreviation)
+
+# List of Figures
+* [DHCPv6 Diagram](#2-modules-design)
+* [DHCPv6 init flow](#4-init-flow)
+
+# Revision
+| Rev | Date     | Author          | Change Description                 |
+|:---:|:--------:|:---------------:|------------------------------------|
+| 0.1 | 03/04    | Shlomi Bitton   | Initial version                    |
+
+# About this Manual
+This document provides an overview of the implementation and integration of DHCP Relay for IPv6 feature in SONiC.
+
+# Scope
+This document describes the high level design of the DHCP Relay for IPv6 feature in SONiC.
+
+# Definitions/Abbreviation
+| Abbreviation  | Description                               |
+|---------------|-------------------------------------------|
+| DHCP          | Dynamic Host Configuration Protocol       |
+
+# 1 Requirements Overview
+
+## 1.1 Functional Requirements
+
+DHCP Relay for IPv6 feature in SONiC should meet the following high-level functional requirements:
+
+- Give the support for relaying DHCP packets from downstream networks to upstream networks using IPv6 addresses.
+- Provide the functionality as a seperate process running on dhcp-relay docker container.
+- Relaying messages to multiple unicast and multicast addresses.
+
+## 1.2 Configuration and Management Requirements
+
+- DHCPv6 trap should be enabled through the COPP manager when the DHCP relay feature is enabled and vice versa. 
+- Downstream network is the VLAN interface with the relay configuration. Global IPv6 address is required to be configured on that interface. 
+- Config DB schema should meet the following format:
+```
+{
```

Comment:
This is not finalized yer, however the idea is to have DHCP in a separate feature table instead of being property of Vlan. Something as follows:

```
    "DHCP": {
        "downlink-intf-i": {
            "dhcpv4_servers": ["dhcp-server-0", "dhcp-server-1", ...., "dhcp-server-n-1"],
            "dhcpv6_servers": ["dhcp-server-0", "dhcp-server-1", ...., "dhcp-server-n-1"]
        }
    }
```

Reply:
Currently it is implemented in a way supervisord on dhcp_relay docker is taking the data by the 'dhcp_servers' on the Vlan interface.
To adapt this new approach all needed is to change the template to this schema.